test/images/agnhost/resource-consumer/consume-cpu/consume_cpu_windows.go

```diff
@@ -0,0 +1,75 @@
+// +build windows
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package consumecpu
+
+import (
+	"syscall"
+	"time"
+
+	syswin "golang.org/x/sys/windows"
+)
+
+type procCPUStats struct {
+	User   int64     // nanoseconds spent in user mode
+	System int64     // nanoseconds spent in system mode
+	Time   time.Time // when the sample was taken
+	Total  int64     // total of all time fields (nanoseconds)
+}
+
+// Retrieves the amount of CPU time this process has used since it started.
+func statsNow(handle syscall.Handle) (s procCPUStats) {
+	var processInfo syscall.Rusage
+	syscall.GetProcessTimes(handle, &processInfo.CreationTime, &processInfo.ExitTime, &processInfo.KernelTime, &processInfo.UserTime)
```

Comment:
I lack sufficient windows expertise to really review this; can we get someone else to take a look at this file?

I am mostly willing to take your word on this one if the tests are happy

Reply:
FWIW, there used to be a problem with this, which has been fixed some time ago: https://github.com/kubernetes-sigs/windows-testing/pull/20

@benmoss could give his OK on this, as he also signed off that change.

Reply:
Yup, this matches the work @astrieanna and I did in the old resource-consumer image.

Reply:
Also, the autoscaling tests that are using the resource-consumer / resource-consumer-controller bits have been added to https://testgrid.k8s.io/sig-windows#aks-engine-azure-windows-master and they're green.